vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -962,6 +993,25 @@ def apply(
         logical_to_physical_map: Optional[torch.Tensor] = None,
         logical_replica_count: Optional[torch.Tensor] = None,
     ) -> torch.Tensor:
+
+        if current_platform.is_xpu():
```

Comment:
cc @bnellnm @tlrmchlsmth @varun-sundar-rabindranath to review the Moe piece, this integration seems pretty intrusive

Reply:
I think ideally there should be a XPU version of `Fp8MoEMethod` that is selected by `Fp8Config.get_quant_method` so that all the XPU specific bits could be isolated to that class.  But a similar kind of dispatching pattern is repeated in `modelopt.py` and `compressed_tensors_moe.py` so I'm not really opposed to this.